core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java

```diff
@@ -795,6 +800,11 @@ public void time() {
     assertEquals(TIME, expr.type());
     assertEquals(new ExprTimeValue("01:01:01"), eval(expr));
     assertEquals("time(TIME '01:01:01')", expr.toString());
+
+    expr = dsl.time(DSL.literal(new ExprTimeValue("2020-01-02 01:01:01")));
```

Comment:
we should test with nano seconds too (separately)

Reply:
Added case for times < 1.0 seconds. https://github.com/Bit-Quill/opensearch-project-sql/pull/134/commits/2016222f524af6c2edd56afa940a53181efd3b1c